src/main/io/gps.c

```diff
@@ -146,6 +146,13 @@ void gpsSetState(gpsState_e state)
     gpsState.lastStateSwitchMs = millis();
 }
 
+static void gpsUpdateTime(void)
+{
+    if (!rtc_has_time() && gpsSol.flags.validTime) {
+        rtc_set_dt(&gpsSol.time);
```

Comment:
I think GPS should call `rtc_set_dt` regardless and let RTC code decide wether it wants to update the time or not.

Reply:
Then we need to change the API to set the time, since we'd have calls which ask to always set the time (e.g. via MSP) and calls that are reporting the time they found from external sources and passing it to the RTC to decide what it wants to do with them (e.g. via GPS).

Reply:
The API should be the same. Maybe `rtcUpdateDateTime` - and RTC system will make a decision if the time should be updated.